modules/bwa/sampe/main.nf

```diff
@@ -0,0 +1,46 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process BWA_SAMPE {
+    tag "$meta.id"
+    label 'process_medium'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    conda (params.enable_conda ? "bioconda::bwa=0.7.17 bioconda::samtools=1.12" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/mulled-v2-fe8faa35dbf6dc65a0f7f5d4ea12e31a79f73e40:66ed1b38d280722529bb8a0167b0cf02f8a0b488-0"
+    } else {
+        container "quay.io/biocontainers/mulled-v2-fe8faa35dbf6dc65a0f7f5d4ea12e31a79f73e40:66ed1b38d280722529bb8a0167b0cf02f8a0b488-0"
+    }
+
+    input:
+    tuple val(meta), path(reads), path(sai)
+    path index
+
+    output:
+    tuple val(meta), path("*.bam"), emit: bam
+    path "*.version.txt"          , emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+    def prefix   = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    def read_group = meta.read_group ? "-r ${meta.read_group}" : ""
+
+    """
+    INDEX=`find -L ./ -name "*.amb" | sed 's/.amb//'`
```

Comment:
is this necessary? Shouldn't `$index` point to that directory anyway?

Reply:
See comment above: https://github.com/nf-core/modules/pull/625#discussion_r680458792

The way `BWA_INDEX` outputs stuff made this more complicated, and I followed `BOWTIE2_ALIGN` conventions

Reply:
Ok, great! Thanks for explaining!